src/ChainlinkOracle.sol

```diff
@@ -51,36 +57,51 @@ contract ChainlinkOracle is IChainlinkOracle {
     /// - The vault's sample shares quoted as assets and the base feed prices don't overflow when multiplied.
     /// - The quote feed prices don't overflow when multiplied.
     /// - The vault, if set, is ERC4626-compliant.
-    /// @param vault Vault. Pass address zero to omit this parameter.
+    /// @param baseVault Base vault. Pass address zero to omit this parameter.
+    /// @param baseVaultConversionSample The sample amount of base vault shares used to convert to the underlying asset.
+    /// Pass 1 if the oracle does not use a vault. Should be chosen such that converting `vaultConversionSample` to
+    /// assets has enough precision.
+    /// @param quoteVault Quote vault. Pass address zero to omit this parameter.
+    /// @param quoteVaultConversionSample The sample amount of quote vault shares used to convert to the underlying
+    /// asset.
+    /// Pass 1 if the oracle does not use a vault. Should be chosen such that converting `vaultConversionSample` to
+    /// assets has enough precision.
     /// @param baseFeed1 First base feed. Pass address zero if the price = 1.
     /// @param baseFeed2 Second base feed. Pass address zero if the price = 1.
     /// @param quoteFeed1 First quote feed. Pass address zero if the price = 1.
     /// @param quoteFeed2 Second quote feed. Pass address zero if the price = 1.
-    /// @param vaultConversionSample The sample amount of vault shares used to convert to the underlying asset.
-    /// Pass 1 if the oracle does not use a vault. Should be chosen such that converting `vaultConversionSample` to
-    /// assets has enough precision.
     /// @param baseTokenDecimals Base token decimals.
     /// @param quoteTokenDecimals Quote token decimals.
     constructor(
-        IERC4626 vault,
+        IERC4626 baseVault,
+        uint256 baseVaultConversionSample,
+        IERC4626 quoteVault,
+        uint256 quoteVaultConversionSample,
         AggregatorV3Interface baseFeed1,
         AggregatorV3Interface baseFeed2,
         AggregatorV3Interface quoteFeed1,
         AggregatorV3Interface quoteFeed2,
-        uint256 vaultConversionSample,
         uint256 baseTokenDecimals,
         uint256 quoteTokenDecimals
```

Comment:
do you like this order, or should we do base-quote

Reply:
i dont really care

Reply:
ok will apply